openeo/extra/artifacts/_s3sts/sts.py

```diff
@@ -27,14 +34,32 @@ def assume_from_openeo_connection(self, connection: Connection) -> AWSSTSCredent
             raise ProviderSpecificException("Only connections that have BearerAuth can be used.")
         auth_token = auth.bearer.split("/")
 
-        return AWSSTSCredentials.from_assume_role_response(
-            self._get_sts_client().assume_role_with_web_identity(
-                RoleArn=self._get_aws_access_role(),
-                RoleSessionName=f"artifact-helper-{Rfc3339().now_utc()}",
-                WebIdentityToken=auth_token[2],
-                DurationSeconds=43200,
+        try:
+            return AWSSTSCredentials.from_assume_role_response(
+                self._get_sts_client().assume_role_with_web_identity(
+                    RoleArn=self._get_aws_access_role(),
+                    RoleSessionName=f"artifact-helper-{Rfc3339().now_utc()}",
+                    WebIdentityToken=auth_token[2],
+                    DurationSeconds=43200,
+                )
             )
-        )
+        except Exception as e:
+            _log.warning("Failed to get credentials for STS access")
+
+            if attempt < 3:
+                # backoff with jitter
+                max_sleep_ms = 500 * (2**attempt)
+                sleep_ms = randint(0, max_sleep_ms)
+                _log.info(f"Retrying STS access in {sleep_ms} ms")
+                sleep(sleep_ms / 1000.0)
+                attempt += 1
+                # Do an API call with OpenEO to trigger a refresh of our token.
+                connection.describe_account()
+                _log.info("Retrying to get credentials for STS access")
+                return self.assume_from_openeo_connection(connection, attempt)
+            else:
+                _log.fatal("Maximum attempts performed for STS access", exc_info=e)
+                raise RuntimeError("Could not get credentials from STS") from e
```

Comment:
doing a "fatal" log and an exception looks a bit like duplication (the exception will likely also be logged at higher level). And "fatal" is maybe a bit much, I don't think we go higher than "error" elsewhere